wallet/src/de/schildbach/wallet/ui/SecurityActivity.kt

```diff
@@ -171,114 +162,81 @@ class SecurityActivity : BaseMenuActivity(), AbstractPINDialogFragment.WalletPro
-            if (configuration.remindBackupSeed){
-                val resetWalletDialog = AdaptiveDialogExt.create(
-                    R.drawable.ic_exclamation_mark_triangle,
-                    getString(R.string.launch_reset_wallet_title),
-                    getString(R.string.launch_reset_wallet_message),
-                    getString(R.string.button_cancel),
-                    getString(R.string.continue_reset),
-                    getString(R.string.launch_reset_wallet_extra_message)
-                )
-                resetWalletDialog.show(this,
-                    onResult = {
-                        if (it == true){
-                            val fiatValue = currentExchangeRate?.coinToFiat(walletBalance)
-                            val startResetWalletDialog = createAdaptiveDialog(
-                                R.drawable.ic_exclamation_mark_triangle,
-                                getString(R.string.start_reset_wallet_title, securityViewModel.getBalanceInLocalFormat(fiatValue)),
-                                getString(R.string.launch_reset_wallet_message),
-                                getString(R.string.button_cancel),
-                                getString(R.string.reset_wallet_text)
-                            )
-                            startResetWalletDialog.show(this){ confirmed ->
-                                if (confirmed == true){
-                                    analytics.logEvent(AnalyticsConstants.Security.RESET_WALLET, bundleOf())
-                                    toAbstractBindService()?.unbindServiceServiceConnection()
-                                    WalletApplication.getInstance().triggerWipe(this)
-                                    startActivity(OnboardingActivity.createIntent(this))
-                                    finishAffinity()
-                                }
+        val walletBalance = viewModel.balance
+        val fiatBalanceStr = viewModel.getBalanceInLocalFormat()
+
+        if (walletBalance.isGreaterThan(Coin.ZERO) && viewModel.needPassphraseBackUp) {
+            val resetWalletDialog = ExtraActionDialog.create(
+                R.drawable.ic_exclamation_mark_triangle,
+                getString(R.string.launch_reset_wallet_title),
+                getString(R.string.launch_reset_wallet_message),
+                getString(R.string.button_cancel),
+                getString(R.string.continue_reset),
+                getString(R.string.launch_reset_wallet_extra_message)
+            )
+            resetWalletDialog.show(this,
+                onResult = {
+                    if (it == true) {
+                        val startResetWalletDialog = AdaptiveDialog.create(
+                            R.drawable.ic_exclamation_mark_triangle,
+                            getString(R.string.start_reset_wallet_title, fiatBalanceStr.ifEmpty {
+                                walletBalance.toFriendlyString()
```

Comment:
If we cannot get a fiat value (no exchange rate available), just show the dash balance